suite-native/module-connect-device/src/components/PinOnDevice.tsx

```diff
@@ -0,0 +1,76 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { Dimensions } from 'react-native';
+import { useCallback, useEffect } from 'react';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { Image, Box, Text } from '@suite-native/atoms';
+import { DeviceModelInternal } from '@trezor/connect';
+import {
+    removeButtonRequests,
+    selectDevice,
+    selectDeviceModel,
+    selectDeviceRequestedPin,
+} from '@suite-common/wallet-core';
+import { Translation } from '@suite-native/intl';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import TrezorConnect, { UI_REQUEST } from '@trezor/connect';
+
+const SCREEN_HEIGHT = Dimensions.get('screen').height;
+
+export const deviceImageMap = {
+    [DeviceModelInternal.T1B1]: require('../assets/pin-t1b1.png'),
+    [DeviceModelInternal.T2T1]: require('../assets/pin-t2t1.png'),
+    [DeviceModelInternal.T2B1]: require('../assets/pin-t2b1.png'),
+} as const satisfies Record<DeviceModelInternal, string>;
+
+const imageStyle = prepareNativeStyle(_ => ({
+    maxHeight: SCREEN_HEIGHT * 0.6,
+    width: 243,
+    height: 550,
+    alignItems: 'center',
+}));
+
+export const PinOnDevice = () => {
+    const dispatch = useDispatch();
+    const navigation = useNavigation();
+
+    const device = useSelector(selectDevice) ?? null;
+    const deviceModel = useSelector(selectDeviceModel);
+    const hasDeviceRequestedPin = useSelector(selectDeviceRequestedPin);
+    const { applyStyle } = useNativeStyles();
+
+    const handleSuccess = useCallback(() => {
+        if (navigation.canGoBack()) {
+            navigation.goBack();
+        }
+        dispatch(removeButtonRequests({ device, buttonRequestCode: 'ButtonRequest_PinEntry' }));
+    }, [navigation, dispatch, device]);
+
+    useEffect(() => {
+        if (!hasDeviceRequestedPin) {
+            handleSuccess();
+        }
+    }, [hasDeviceRequestedPin, handleSuccess]);
+
+    useEffect(() => {
+        TrezorConnect.on(UI_REQUEST.CLOSE_UI_WINDOW, handleSuccess);
+
+        return () => TrezorConnect.off(UI_REQUEST.CLOSE_UI_WINDOW, handleSuccess);
+    }, [handleSuccess]);
+
+    if (!deviceModel) return null;
+
+    return (
+        <Box flex={1} justifyContent="space-between" alignItems="center" paddingTop="large">
```

Comment:
please, make `style={applyStyle...}` for this

Reply:
83ace625ed5bf6b070f5714dab15cba7ff17bac0

Reply:
Please, can you also add `paddingTop` to the style?

Reply:
I was confused by types 
<img width="1203" alt="image" src="https://github.com/trezor/trezor-suite/assets/3729633/01dd4aea-1960-4089-9ee2-61c959a677cc">
but it helps me to find out how to use it, so thanks for pointing that out again 🙏 
22468226e3

Reply:
perfect 👌